src/jitlayers.cpp

```diff
@@ -1240,18 +1240,46 @@ void jl_dump_native(const char *bc_fname, const char *obj_fname, const char *sys
     PM.add(new DataLayout(*jl_ExecutionEngine->getDataLayout()));
 #endif
 
-    addOptimizationPasses(&PM);
-
+    std::unique_ptr<raw_fd_ostream> unopt_bc_OS;
     std::unique_ptr<raw_fd_ostream> bc_OS;
     std::unique_ptr<raw_fd_ostream> obj_OS;
 #if JL_LLVM_VERSION >= 30700 // 3.7 simplified formatted output; just use the raw stream alone
+    std::unique_ptr<raw_fd_ostream> &unopt_bc_FOS = unopt_bc_OS;
     std::unique_ptr<raw_fd_ostream> &bc_FOS = bc_OS;
     std::unique_ptr<raw_fd_ostream> &obj_FOS = obj_OS;
 #else
+    std::unique_ptr<formatted_raw_ostream> unopt_bc_FOS;
     std::unique_ptr<formatted_raw_ostream> bc_FOS;
     std::unique_ptr<formatted_raw_ostream> obj_FOS;
 #endif
 
+
+    if (unopt_bc_fname) {
+#if JL_LLVM_VERSION >= 30500
+        // call output handler directly to avoid special case handling of `-` filename
+        int FD;
+        std::error_code EC = sys::fs::openFileForWrite(unopt_bc_fname, FD, sys::fs::F_None);
+        unopt_bc_FOS.reset(new raw_fd_ostream(FD, true));
+        std::string err;
+        if (EC)
+            err = "ERROR: failed to open --output-bc file '" + std::string(unopt_bc_fname) + "': " + EC.message();
+#else
+        std::string err;
+        unopt_bc_OS.reset(new raw_fd_ostream(unopt_bc_fname, err, raw_fd_ostream::F_Binary));
+#endif
+        if (!err.empty())
+            jl_safe_printf("%s\n", err.c_str());
+        else {
+#if JL_LLVM_VERSION < 30700
+            unopt_bc_FOS.reset(new formatted_raw_ostream(*unopt_bc_OS.get()));
+#endif
+            PM.add(createBitcodeWriterPass(*unopt_bc_FOS.get()));     // Unroll small loops
```

Comment:
Ah yeah, that's been there forever. Guess I might as well fix that while I'm here.